src/WebJobs.Script.WebHost/Security/Authentication/Keys/AuthenticationLevelHandler.cs

```diff
@@ -47,13 +54,17 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
 
             List<ClaimsIdentity> claimsIdentities = new List<ClaimsIdentity>();
 
-            if (_isEasyAuthEnabled)
+            if (_isEasyAuthEnabled
+                && Context.Request.TryGetAppServiceIdentity(out ClaimsIdentity easyAuthIdentity)
+                && easyAuthIdentity != null)
```

Comment:
This should in theory never be `null` since we have already tested that `TryGetAppServiceIdentity` is true, but it probably doesn't hurt anything to leave it in here.